src/Columns/ColumnAggregateFunction.cpp

```diff
@@ -330,7 +330,38 @@ ColumnPtr ColumnAggregateFunction::filter(const Filter & filter, ssize_t result_
 
 void ColumnAggregateFunction::expand(const Filter & mask, bool inverted)
 {
-    expandDataByMask<char *>(data, mask, inverted);
```

Comment:
isn't it the case for `expandDataByMask<char *>(...)` in general? we have a general instantiation of this function:
https://github.com/Avogar/ClickHouse/blob/variant-aggregate-column/src/Columns/MaskOperations.cpp#L73
shouldn't me specialize this instantiation with your solution instead?

Reply:
Here is a bit different situation. We must allocate each default value in the arena, we cannot just provide a default value in `expandDataByMask`. We could change `expandDataByMask` and change the default value to the function that creates default value, but I think it's not a good idea for general case

Reply:
I tried to allocate single default value in advance and use it, but it doesn't work, this column cannot have the same value in different rows (I didn't fully understand why). So, we must allocate each default value in the arena before insertion.

Reply:
>I tried to allocate single default value in advance and use it, but it doesn't work

I think it's because it's assumed that value for every position is separate - I see something like `func->destroy(data[i]);` everywhere...